src/sentry/snuba/metrics/query_builder.py

```diff
@@ -327,39 +277,17 @@ def get_date_range(params: Mapping) -> Tuple[datetime, datetime, int]:
     interval = parse_stats_period(params.get("interval", "1h"))
     interval = int(3600 if interval is None else interval.total_seconds())
 
-    # hard code min. allowed resolution to 10 seconds
-    allowed_resolution = AllowedResolution.ten_seconds
-
-    smallest_interval, interval_str = allowed_resolution.value
-    if interval % smallest_interval != 0 or interval < smallest_interval:
-        raise InvalidParams(
-            f"The interval has to be a multiple of the minimum interval of {interval_str}."
-        )
-
-    if ONE_DAY % interval != 0:
-        raise InvalidParams("The interval should divide one day without a remainder.")
-
     start, end = get_date_range_from_params(params, default_stats_period=timedelta(days=1))
+    date_range = timedelta(
+        seconds=int(interval * math.ceil((end - start).total_seconds() / interval))
```

Comment:
it would be great to extract this expression to func, so we can name it somehow to explain what it does.

Reply:
agreed